hellabot.go

```diff
@@ -353,11 +313,60 @@ type Trigger struct {
 // client has timed out and will close the connection.
 // Note: this is automatically added in the IrcCon constructor
 var pingPong = &Trigger{
-	func(m *Message) bool {
+	func(bot *Bot, m *Message) bool {
 		return m.Command == "PING"
 	},
-	func(irc *IrcCon, m *Message) bool {
-		irc.Send("PONG :" + m.Content)
+	func(bot *Bot, m *Message) bool {
+		bot.Send("PONG :" + m.Content)
 		return true
 	},
 }
+var joinChannels = &Trigger{
+	func(bot *Bot, m *Message) bool {
+		return m.Command == irc.RPL_WELCOME // || m.Command == irc.RPL_ENDOFMOTD // 001 or 372
```

Comment:
Because a server replies with both in most cases, but some only give one of them. Needs a sync.Once to avoid trying to join twice. 

On August 30, 2015 12:46:02 AM CEST, Lars Gierth <notifications@github.com> wrote:
>>  		return true
>>  	},
>>  }
>> +var joinChannels = &Trigger{
>> +	func(bot *Bot, m *Message) bool {
>> +		return m.Command == irc.RPL_WELCOME // || m.Command ==
>irc.RPL_ENDOFMOTD // 001 or 372
>
>Why is RPL_ENDOFMOTD commented out?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/whyrusleeping/hellabot/pull/16/files#r38266714

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.